src/lib/auth.ts

```diff
@@ -91,12 +90,12 @@ export const DEVICE_API_KEY_PERMISSIONS = [
 	`resin.device.create?belongs_to__application/any(a:a/depends_on__application/any(da:da/${ownsDevice}))`,
 	`resin.device.update?belongs_to__application/any(a:a/depends_on__application/any(da:da/${ownsDevice}))`,
 
-	'resin.service.read?application/canAccess()',
+	'resin.service.read?application/canAccess() or service_install/canAccess()',
```

Comment:
I think you can probably ditch the `application/canAccess()`?
```suggestion
	'resin.service.read?service_install/canAccess()',
```

Reply:
@Page- I think that we can't since if we do so, then when a device is patching its own `should_be_running__release` then the `createReleaseServiceInstalls()` will not be able to retrieve the services for which it needs to create the service installs for (except of course if we change that to use root permissions).

See: https://github.com/balena-io/open-balena-api/blob/master/src/features/ci-cd/hooks/service-installs.ts#L11-L62

See: https://github.com/balena-io/open-balena-api/pull/877

Reply:
@Page- even with the `resin.service.read?application/canAccess() or service_install/canAccess()`, I think the state endpoint query is quite close to the fully optimized one you ended up:
[device state perms PR 874.txt](https://github.com/balena-io/open-balena-api/files/7621876/device.state.perms.PR.874.txt)

The only diff I could spot was dependent devices related and had nothing to do with services.

Reply:
Removing the branch `application/canAccess()` won't add any optimizations (it'll actually just remove them if going `application -> service`), I just thought it'd probably be unnecessary with the addition of the `service_install/canAccess()`, but you're right I hadn't considered that the service installs need to be added somehow

Reply:
I have to say that the way of how cleverly the extra permission checks are omitted in this `application/canAccess() or service_install/canAccess()` case amazed me while I was giving this a try yesternight.